include/os/linux/spl/sys/kmem_cache.h

```diff
@@ -118,9 +118,6 @@ extern struct rw_semaphore spl_kmem_cache_sem;
 #define	SPL_MAX_KMEM_ORDER_NR_PAGES	(KMALLOC_MAX_SIZE >> PAGE_SHIFT)
 #endif
 
-#define	POINTER_IS_VALID(p)		0	/* Unimplemented */
-#define	POINTER_INVALIDATE(pp)			/* Unimplemented */
```

Comment:
Removing this debugging makes sense to me.  We _could_ consider going farther and move `dnode_move()` in to the platform specific code.  This functionality is not implement on Linux, and it sounds like it's not used on FreeBSD or OSX either.  Only Illumos makes use of it.

Reply:
Moving is fine by me.
`dnode_move()` we have on osx/windows as we use illumos kmem. The one we can not do on osx is `znode_move()` as we can not update `vp->v_data`.

Reply:
Ahh OK I see.  Well then, if we do have multiple consumers and it's not causing any real trouble let's go ahead and keep it in the common code.  Thanks.

Reply:
```
/*
 * Helps clients implementing the move() callback to recognize known objects by
 * testing a client-designated pointer member. Takes advantage of the fact that
 * any scribbling to freed memory done by kmem is guaranteed to set one of the
 * two low order bits.
 */
#define POINTER_IS_VALID(p)     (!((uintptr_t)(p) & 0x3))
#define POINTER_INVALIDATE(pp)  (*(pp) = (void *)((uintptr_t)(*(pp)) | 0x1))
```
we are using on illumos/dilos, for testing of this solution i need a time, but it'll be more visible a little later to me and you can do updates if you need it now.